Fullstory integration Update.

LCOleksii opened the pull request:
### Details
- FS Pod dependency updated to 1.48
- React-Native FS lib updated to 1.4.2
- Video preview, Attachment Preview fs-exclude class added to exclude session recording.
- Cleanup redundant fsClass="fs-exclude" and props


### Fixed Issues
$ https://github.com/Expensify/App/issues/40910
$ https://github.com/Expensify/App/issues/40784



### Tests
1. Click on the attachment upload menu
3. Upload an Video via file manager selection
4. Verify a modal appears displaying a preview of the Video
5. Upload the Attachment file

- [x] Verify that no errors appear in the JS console


### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
1. Login to the App 
2. Open chat
3. Click on the attachment upload menu
4. Upload an Video via file manager selection
6. Verify a modal appears displaying a preview of the Video
7. Upload the Attachment file

- [x] Verify that app does not crash
- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>
https://drive.google.com/file/d/1EQRkfkWRo5Gm8uBQ_nSJg-h5XqDBJwqB/view?usp=drive_link
</details>

<details>
<summary>iOS: Native</summary>
https://drive.google.com/file/d/1R3ALatTtvHBW-E0x4PqhbcUrVv70s1UC/view?usp=drive_link
</details>

<details>
<summary>MacOS: Chrome / Safari</summary>
https://drive.google.com/file/d/1pK0UacdVlrCWuSCSLpsrGs7rHsjiGu-N/view?usp=drive_link
</details>

<details>
<summary>MacOS: Desktop</summary>
https://drive.google.com/file/d/1m2DdKWPDjrY7HAVVhHeRLZNhgmGlSVAq/view?usp=drive_link
</details>

danieldoglas committed:
Merge pull request #40865 from Expensify/revert-35226-fix/34442

[CP Staging] Revert "Implement suggestion for edit composer"

(cherry picked from commit 95aa19a5a3f795a4d5bcec135188b765073b0a37)

OSBotify committed:
Update version to 1.4.65-1

(cherry picked from commit e58ec4940cbed9352266760bebee0a03257c4717)

mountiny committed:
Merge pull request #40807 from Expensify/dsilva_changingPropertyBeingUsedByFullstory

[CP Staging] Changing properties being used by Fullstory

(cherry picked from commit 51372ab1171059ff514ffed1dc1511452aee6a57)

mountiny committed:
Merge pull request #40807 from Expensify/dsilva_changingPropertyBeingUsedByFullstory

[CP Staging] Changing properties being used by Fullstory

(cherry picked from commit 51372ab1171059ff514ffed1dc1511452aee6a57)

OSBotify committed:
Update version to 1.4.65-2

(cherry picked from commit 390bf24ff2b98059b2527e40931fdac3f595d31e)

LCOleksii committed:
Merge branch 'main' into fullstory-integration

# Conflicts:
#	android/app/build.gradle
#	ios/NewExpensify/Info.plist
#	ios/NewExpensifyTests/Info.plist
#	ios/NotificationServiceExtension/Info.plist
#	package-lock.json
#	package.json

LCOleksii committed:
Changes:
- IOs FullStory pod updated to version "1.47.0"
- babel-plugin-react-native version updated to version "1.2.1"
- Video preview, Attachment Preview fs-exclude class added

LCOleksii committed:
Merge branch 'main' into fullstory-integration

# Conflicts:
#	android/app/build.gradle
#	android/build.gradle
#	babel.config.js
#	ios/NewExpensify.xcodeproj/project.pbxproj
#	ios/NewExpensify/Info.plist
#	ios/NewExpensifyTests/Info.plist
#	ios/NotificationServiceExtension/Info.plist
#	ios/Podfile.lock
#	package-lock.json
#	package.json

LCOleksii committed:
Changes:
- Latest fixes

LCOleksii committed:
Changes:
- Cleanup redundant fsClass="fs-exclude" and props
- FS Pod dependency updated to 1.48
- React-Native FS lib updated to 1.4.2

LCOleksii committed:
Merge branch 'main' into fullstory-integration

LCOleksii committed:
Changes:
- Cleanup eslint/prettier
- Added Fullstory lib to TSX (View Properties extend attributes)

danieldoglas commented:
@alitoshmatov , removing you from this one since @ishpaul777 was already assigned to the original issue!

LCOleksii committed:
Changes:
- Cleanup eslint/prettier
- Merge conflicts rollback deleted folder/files
- Added Fullstory lib to TSX (View Properties extend attributes)

danieldoglas commented:
@LCOleksii can you please apply this PR on top of yours? https://github.com/Expensify/App/pull/40807/files

We've implemented the user metadata keys, so now you can use those to map everything instead of the session key. That was reverted with the other changes.

LCOleksii committed:
Changes:
- Cleanup eslint/prettier
- Added TSX fullstory file to trigger types overrides

LCOleksii committed:
Merge branch 'main' into fullstory-integration

LCOleksii committed:
Merge branch 'main' into fullstory-integration

LCOleksii committed:
Changes:
- Fixes eslint/prettier

danieldoglas commented:
Looks like we're still missing this: 
> can you please apply this PR on top of yours? https://github.com/Expensify/App/pull/40807/files

LCOleksii committed:
Merge branch 'main' into fullstory-integration

danieldoglas commented:
@ishpaul777 theres one more commit incoming, so please wait until that's done!

LCOleksii commented:
Hi @ishpaul777 
Cannot reproduce the issue, also prior last commit push i have checked locally.
Please Insure that you had:
- Had clean the project ('npm run clean') + node_modules remove (to be sure no cache issues, we had similar issues that could be fixed only after proper clean, especially if you are switching between the branches)
- Reinstalled all the pods for IOs build
- Deleted previous build from the IOs simulator
- Build the new version from the branch

Please see the latest recording:

[Simulator Screen Recording - iPhone 15 Plus - 2024-05-20 at 22.37.25](https://drive.google.com/file/d/1ReBxVM5YNYQU09tfiGfF1_zjzi3j3BqO/view?usp=drive_link)

<img width="1396" alt="Screenshot 2024-05-20 at 23 17 13" src="https://github.com/Expensify/App/assets/94551035/1071a430-3fa1-48cd-8f90-1505d2f3c747">

If after this the issue still persist, please could you provide full crash log for investigation.

ishpaul777 reviewed (commented):
Approving since adhoc works fine 🤷

kbecciv commented:
## Attachment - Corrupted pdf can be uploaded without error

**Version Number:** v1.4.74-4
**Reproducible in staging?:** n
**Reproducible in production?:** n

## Action Performed:
1. Launch New Expensify app.
2. Go to any chat.
3. Go to + > Add attachment > Choose document.
4. Select a corrupted pdf file (attached below).
5. Upload the corrupted file.

## Expected Result:
In Step 4, the preview will show "Failed to load PDF file" (production behavior).
In Step 5, the corrupted file will not be uploaded and shows "Failed to load PDF file" (production behavior).

## Actual Result:
In Step 4, the preview shows filename.pdf.
In Step 5, the corrupted file is uploaded successfully.

## Workaround:
n/a

## Platforms:
<!---
Check off any platforms that are affected by this issue
--->
Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] iOS: Native

## Screenshots/Videos

Add any screenshot/video evidence


https://github.com/Expensify/App/assets/93399543/f611b5c0-2b17-4e66-997e-ccfae13853bd


https://github.com/Expensify/App/assets/93399543/934cd551-51a5-4b02-b211-17cf03ce6f92

danieldoglas committed:
adding further changes

danieldoglas commented:
@ishpaul777 I've done a few other changes, can you please recheck/approve? this will only affect loading the fullstory lib in production, so you don't need to do all tests again

danieldoglas committed:
prettier

OSBotify commented:
🚀 [Cherry-picked](https://github.com/Expensify/App/actions/runs/9258352492) to staging by https://github.com/puneetlath in version: 1.4.76-1 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

@Expensify/applauseleads please QA this PR and check it off on the [deploy checklist](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3AStagingDeployCash) if it passes.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9283349442) to production by https://github.com/Beamanator in version: 1.4.76-7 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
